### PR TITLE
[Meteostick] Add windvane calibration

### PR DESCRIPTION
**Pull request by Chiuaua79**
Ref. issue [16268](https://github.com/openhab/openhab-addons/issues/16268)

The Davis wind vanes 6410, 7911 and 7914 are
mounted to a flat surface on the shaft. To ensure
correct orientation of the wind vane, the
anemometer must be mounted so that the arm points
true north. This is not always possible. In these
cases the wind direction must be calibrated. This
commit adds the option to enter the direction the
arm is facing and the binding will recalculate the wind direction as received from the Meteostick
before publishing to the relevant channel.

Furthermore in my opinion one rain channel was
missing: accumulative rainfall for the current
date. Added this channel, which required some code changes (adding of cronScheduler so the value is
reset at midnight of the selected timezone). Tried to improve the code.

### CODE REVIEW

File: bundles/org.openhab.binding.meteostick/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -79,6 +79,13 @@
 				<default>0.254</default>
 				<unitLabel>mm</unitLabel>
 			</parameter>
+
+			<parameter name="deltawinddir" type="integer" required="false" min="0" max="359">
+				<label>Wind vane direction</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.meteostick/src/main/resources/OH-INF/thing/thing-types.xml:84]: Please capitalize this label adn re-run i18n plugin.

File: bundles/org.openhab.binding.meteostick/src/main/java/org/openhab/binding/meteostick/internal/handler/MeteostickSensorHandler.java [hunk 2]
```diff
@@ -79,7 +81,11 @@ public void initialize() {
         if (spoon == null) {
             spoon = new BigDecimal(PARAMETER_SPOON_DEFAULT);
         }
-        logger.debug("Initializing MeteoStick handler - Channel {}, Spoon size {} mm.", channel, spoon);
+
+        deltawinddir = ((BigDecimal) getConfig().get(PARAMETER_WINDVANE)).intValue();
```
Comment (lsiepel) [inline bundles/org.openhab.binding.meteostick/src/main/java/org/openhab/binding/meteostick/internal/handler/MeteostickSensorHandler.java:85]:
```suggestion
        deltawinddir = (int) getConfig().get(PARAMETER_WINDVANE);
```

File: bundles/org.openhab.binding.meteostick/src/main/java/org/openhab/binding/meteostick/internal/MeteostickBindingConstants.java [hunk 3]
```diff
@@ -51,6 +51,7 @@ public class MeteostickBindingConstants {
     public static final String PARAMETER_CHANNEL = "channel";
     public static final String PARAMETER_SPOON = "spoon";
     public static final String PARAMETER_SPOON_DEFAULT = "0.254";
+    public static final String PARAMETER_WINDVANE = "deltawinddir";
```
Comment (lsiepel) [inline bundles/org.openhab.binding.meteostick/src/main/java/org/openhab/binding/meteostick/internal/MeteostickBindingConstants.java:54]: Please use camelCase: https://www.openhab.org/docs/developer/guidelines.html#naming-convention
```suggestion
    public static final String PARAMETER_WINDVANE = "deltaWindDirection";
```

Also update the readme and xml.

File: bundles/org.openhab.binding.meteostick/README.md [hunk 4]
```diff
@@ -41,10 +41,11 @@ Set mode to one of the following depending on your device and region:
...
 
-| Option  | Description                               |
-|---------|-------------------------------------------|
-| channel | Sets the RF channel used for this sensor  |
-| spoon   | Size of rain spoon assembly for this sensor in mm.  Default value is 0.254 (0.01") for use with Davis part number 7345.280.  Set to 0.2 for use with Davis part number 7345.319 |
+| Option       | Description                               |
+|--------------|-------------------------------------------|
+| channel      | Sets the RF channel used for this sensor  |
+| spoon        | Size of rain spoon assembly for this sensor in mm.  Default value is 0.254 (0.01") for use with Davis part number 7345.280.  Set to 0.2 for use with Davis part number 7345.319 |
+| deltawinddir | For Davis 6410, 7911 & 7914 anemometers, if your anemometer cannot be mounted aiming true North set the direction it is aiming here (0 to 359 degrees). Default is 0 (for North) |
```
Comment (lsiepel) [inline bundles/org.openhab.binding.meteostick/README.md:48]:
```suggestion
| Option       | Description                                                                                                                                                                      |
|--------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| channel      | Sets the RF channel used for this sensor                                                                                                                                         |
| spoon        | Size of rain spoon assembly for this sensor in mm.  Default value is 0.254 (0.01") for use with Davis part number 7345.280.  Set to 0.2 for use with Davis part number 7345.319  |
| deltawinddir | For Davis 6410, 7911 & 7914 anemometers, if your anemometer cannot be mounted aiming true North set the direction it is aiming here (0 to 359 degrees). Default is 0 (for North) |
```

**Review by lsiepel — CHANGES_REQUESTED** [review]
Left some minor comments

File: bundles/org.openhab.binding.meteostick/src/main/java/org/openhab/binding/meteostick/internal/handler/MeteostickSensorHandler.java (hunk 2 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.meteostick/src/main/java/org/openhab/binding/meteostick/internal/handler/MeteostickSensorHandler.java:93]: Hmm. didn't expect that. After some checks i conclude i was wrong, sorry.

**Review by lsiepel — APPROVED** [review]
LGTM

**Comment by Chiuaua79** [comment]
Had to push once more as apparently the channel for rain-today was not showing up in the binding due to missing info in OH-INF/i18n/meteostick.properties and OH-INF/thing/thing-types.xml. Fixed these files as well to show the new channel (tested in my production OH server at home).

File: bundles/org.openhab.binding.meteostick/src/main/resources/OH-INF/thing/thing-types.xml [hunk 5]
```diff
@@ -190,6 +198,15 @@
 		</state>
 	</channel-type>
 
+	<channel-type id="rain-today">
+		<item-type>Number:Length</item-type>
+		<label>Rainfall (today)</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.meteostick/src/main/resources/OH-INF/thing/thing-types.xml:203]:
```suggestion
		<label>Rainfall Today</label>
```

please also re-run i18n plugin

**Review by lsiepel — CHANGES_REQUESTED** [review]
Minor label adjustment.

File: bundles/org.openhab.binding.meteostick/src/main/resources/OH-INF/thing/thing-types.xml (hunk 5 above)
Comment (Chiuaua79) [reply bundles/org.openhab.binding.meteostick/src/main/resources/OH-INF/thing/thing-types.xml]: Dear @lsiepel,

I committed the suggested change via the ipad app, but due to the commit message not being according openhab specifics (signing off with name and valid e-mail address) I had to perform a rebase and a reword of the commit message on my Ubuntu PC using the terminal (command line). So far so good.

What I don't understand is your remark "please also re-run i18n plugin". I tried to google and search in the forum to check what I have to do exactly... but to no avail.

Comment (lsiepel) [reply bundles/org.openhab.binding.meteostick/src/main/resources/OH-INF/thing/thing-types.xml]: Documentation can be found here: https://www.openhab.org/docs/developer/utils/i18n.html#generating-i18n-properties-file

It is needed as the cvhange in the XML has to be reflexted in the .properties file.

**Review by lsiepel — APPROVED** [review]
LGTM, thanks

**Pull request merged**